src/main/java/agh/ics/oop/GrassField.java

```diff
@@ -0,0 +1,90 @@
+package agh.ics.oop;
+
+import java.util.*;
+
+public class GrassField extends AbstractWorldMap {
+
+    private final ArrayList<Vector2d> garden;
+
+    public GrassField(int grassCount) {
+        super();
+        this.garden = new ArrayList<>();
+        int MAP_BOUNDARY = (int) Math.sqrt(10 * grassCount);
+        for (int i = 0; i < MAP_BOUNDARY; i++) {
+            for (int j = 0; j < MAP_BOUNDARY; j++) {
+                this.garden.add(new Vector2d(i, j));
+            }
+        }
+        this.putGrass(grassCount);
+    }
+
+    @Override
+    public boolean place(Animal animal) {
+        Object object = this.objectAt(animal.getPosition());
+        if (object instanceof Grass) {
+            garden.remove(animal.getPosition());
+            this.elements.remove(object);
+            putGrass(1);
```

Comment:
ok, przy czym to zadziała tylko przy wstawianiu zwierzątka, przy jego przesuwaniu już nie.